src/hotspot/share/runtime/os.cpp

```diff
@@ -2064,6 +2064,19 @@ static void assert_nonempty_range(const char* addr, size_t bytes) {
          p2i(addr), p2i(addr) + bytes);
 }
 
+julong os::used_memory() {
+#ifndef LINUX
+  if (OSContainer::is_containerized()) {
+    jlong mem_usage = OSContainer::memory_usage_in_bytes();
+    if (mem_usage > 0) {
+      return mem_usage;
+    }
+  }
```

Comment:
I'm not sure we should introduce Linux specific things in the shared os code like that. It seems the model is to use `pd_<foo>` to delegate to platform specific code? @dholmes-ora thoughts?

Reply:
I think that is okay here. Sometimes its the lesser of two evils.

I actually understood David's remark that way, too (https://github.com/openjdk/jdk/pull/19772#pullrequestreview-2127842338)

Reply:
I understood the remark as "reduce the default implementation duplication". I.e. not have the same impl for all platforms. A pd_<foo> would just move the impl to `os/linux` rather than having it in `share/runtime`. But if that's fine, then I won't object, of course.

Reply:
I am okay with the proposed code. If it were larger then I'd want it hidden behind a pd function but this is okay.